src/function/table/table_scan.cpp

```diff
@@ -64,16 +67,24 @@ static unique_ptr<LocalTableFunctionState> TableScanInitLocal(ExecutionContext &
 		auto storage_idx = GetStorageIndex(*bind_data.table, col);
 		col = storage_idx;
 	}
+
 	result->scan_state.table_filters = input.filters;
 	TableScanParallelStateNext(context.client, input.bind_data, result.get(), gstate);
 	return move(result);
 }
 
 unique_ptr<GlobalTableFunctionState> TableScanInitGlobal(ClientContext &context, TableFunctionInitInput &input) {
+
 	D_ASSERT(input.bind_data);
 	auto &bind_data = (const TableScanBindData &)*input.bind_data;
 	auto result = make_unique<TableScanGlobalState>(context, input.bind_data);
+
 	bind_data.table->storage->InitializeParallelScan(context, result->state);
+	// acquire index scan specific locks
+	if (bind_data.is_create_index) {
+		bind_data.table->storage->InitializeCreateIndexScan(result->create_index_scan_state, input.column_ids);
```

Comment:
Looks like you `InitializeParallelScan` as well as `InitializeCreateIndexScan` here, which seems redundant.

Can you perhaps only call one of these, and change `InitializeCreateIndexScan` to initialize a parallel scan instead of a single-threaded scan?